sphinx/domains/c/__init__.py

```diff
@@ -666,6 +667,10 @@ class CDomain(Domain):
         'objects': {},  # fullname -> docname, node_id, objtype
     }
 
+    initial_intersphinx_inventory = {
```

Comment:
```suggestion
    initial_intersphinx_inventory: dict[str, Any] = {
```

Do you think we should only store the symbol (even for domains subclassing the C-domain?) or do you think we should allow a larger type (in which the subclasses would need to fix probably everything?)

Reply:
I have changed it to the concrete type ``dict[str, Symbol]`` now. Indeed, if someone wants to subclass the domain and change this dictionary, they probably need to change everything else related to the delegation.